rust/xaynet-server/src/storage/redis.rs

```diff
@@ -258,7 +197,60 @@ impl Connection {
         Ok(seed_dict)
     }
 
+    /// Updates the [`SeedDict`] with the seeds from the given ['UpdateParticipantPublicKey'].
+    pub async fn update_seed_dict(
+        mut self,
+        update_pk: &UpdateParticipantPublicKey,
+        local_seed_dict: &LocalSeedDict,
+    ) -> RedisResult<SeedDictUpdate> {
+        let script = Script::new(
+            r#"
+                local update_pk = ARGV[1]
+
+                -- check if the local seed dict has the same length as the sum_dict
+
+                -- KEYS is a list (table) of key value pairs ([update_pk_1, seed_1, update_pk_2, seed_2])
+                local seed_dict_len = #KEYS / 2
+                local sum_dict_len = redis.call("HLEN", "sum_dict")
+                if seed_dict_len ~= sum_dict_len then
+                    return -1
+                end
+
+                -- check if all pks of the local seed dict exists in sum_dict
+                for i = 1, #KEYS, 2 do
+                    local exist_in_sum_dict = redis.call("HEXISTS", "sum_dict", KEYS[i])
+                    if exist_in_sum_dict == 0 then
+                        return -2
+                    end
+                end
+
+                -- check if one pk of the local seed dict already exists in seed_dict
+                local exist_in_seed_dict = redis.call("SADD", "update_participants", update_pk)
+                -- SADD returns 0 if the key already exists
+                if exist_in_seed_dict == 0 then
+                    return -3
+                end
+
+                -- update the seed dict
+                for i = 1, #KEYS, 2 do
+                    redis.call("HSETNX", KEYS[i], update_pk, KEYS[i + 1])
```

Comment:
Although this _shouldn't_ fail, maybe we could check for an error here as well? Even though I don't really know how we'd handle it, I think it would be nice to just log it, so that we know it happened.

Reply:
Good point👍 I will add that